tests/test_trainer.py

```diff
@@ -93,6 +110,46 @@ def test_column_mapping_with_missing_text(self):
         with pytest.raises(ValueError):
             trainer._validate_column_mapping(trainer.train_dataset)
 
+    def test_column_mapping_raises_warning_when_mapped_columns_missing(self):
+        """If the columns specified in the column mapping are missing from the dataset, the library should log a warning and list the columns found."""
+        dataset = Dataset.from_dict({"text": ["a", "b", "c"], "extra_column": ["d", "e", "f"]})
+        trainer = SetFitTrainer(
+            model=self.model,
+            train_dataset=dataset,
+            eval_dataset=dataset,
+            num_iterations=self.num_iterations,
+            column_mapping={"text_new": "text", "label_new": "label"},
+        )
+        with self.assertLogs(level=logging.WARNING) as cm:
+            try:
+                trainer._validate_column_mapping(trainer.train_dataset)
+            except ValueError:
+                pass
+            self.assertIn(
+                "WARNING:setfit.trainer:The column mapping looks for the columns ['label_new', 'text_new'] in the dataset, "
+                "but the dataset has the columns ['extra_column', 'text'].",
+                cm.output,
+            )
+
+    def test_trainer_raises_warning_when_dataset_not_split(self):
+        """Verify that a warning is raised if we pass an unsplit dataset to the trainer."""
+        dataset = Dataset.from_dict({"text": ["a", "b", "c", "d"], "label": [0, 0, 1, 1]}).train_test_split(
+            test_size=0.5
+        )
+        trainer = SetFitTrainer(
+            model=self.model, train_dataset=dataset, eval_dataset=dataset, num_iterations=self.num_iterations
+        )
+        with self.assertLogs(level=logging.WARNING) as cm:
+            try:
+                trainer._validate_column_mapping(trainer.train_dataset)
+            except ValueError:
+                pass
```

Comment:
```suggestion
            with self.assertRaises(ValueError):
                trainer._validate_column_mapping(trainer.train_dataset)
```
Idem

Reply:
Fixed in [97232d1](https://github.com/huggingface/setfit/pull/299/commits/97232d18f587c8d87cf7d806f9efd0e285b3b0ab)